### PR TITLE
refine: cache lodGroup's state and reduce api call of getVisibleLODLevel

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/pipeline/LODModelsUtil.cpp [hunk 1]
```diff
@@ -33,6 +33,12 @@
 namespace cc {
 namespace pipeline {
 
+struct LODInfo {
+    int8_t visibleLevel{-1};
+    bool needUpdate{true};
+    Node::AncestorTransformChanged::EventID eventId;
```
Comment (minggo) [inline native/cocos/renderer/pipeline/LODModelsUtil.cpp]: default value for eventId

File: native/cocos/renderer/pipeline/LODModelsUtil.cpp [hunk 2]
```diff
@@ -43,15 +49,52 @@ ccstd::unordered_map<const scene::Model *, bool> modelsInAnyLODGroup;
...
+ * @zh 指定相机下，某个LODGroup的当前状态，如：当前使用哪级的LOD，是否需要更新等
+ * @en Specify the current status of a LODGroup under the camera, such as: which level of LOD is currently in use, whether it needs to be updated, etc.
+ */
+ccstd::unordered_map<const scene::Camera *, ccstd::unordered_map<const scene::LODGroup *, LODInfo>> lodGroupStateMap;
+
+/**
+ * @zh 指定相机的状态是否出现变化, 这里主要记录transform是否出现变化
+ * @en Specify whether the state of the camera has changed or not, here it is mainly recorded whether the transform has changed or not.
+ */
+ccstd::unordered_map<const scene::Camera *, std::tuple<bool, Node::AncestorTransformChanged::EventID>> cameraStateMap;
```
Comment (minggo) [inline native/cocos/renderer/pipeline/LODModelsUtil.cpp]: Please define them in anonymous namespace

File: native/cocos/renderer/pipeline/LODModelsUtil.cpp [hunk 3]
```diff
@@ -43,15 +49,52 @@ ccstd::unordered_map<const scene::Model *, bool> modelsInAnyLODGroup;
...
+            [camera](cc::Node * /* emitter*/, cc::TransformBit  /*transformBit*/) {
+                std::get<0>(cameraStateMap[camera]) = true;
+            });
+    }
+    auto &visibleModels = visibleModelsByAnyLODGroup[camera];
+    auto &lodInfoMap = lodGroupStateMap[camera];
+
     for (const auto &lodGroup : scene->getLODGroups()) {
         if (lodGroup->isEnabled()) {
+            bool lodInfoExist = lodInfoMap.find(lodGroup) != lodInfoMap.end();
```
Comment (minggo) [inline native/cocos/renderer/pipeline/LODModelsUtil.cpp]: use `lodInfoMap.count(lodGroup)` has better performance

**Review by stanleyljl — APPROVED** [review]

File: cocos/rendering/custom/web-scene.ts [hunk 4]
```diff
@@ -110,16 +109,12 @@ export class WebSceneTask implements SceneTask {
...
-            LODModelsCachedUtils.updateCachedLODModels(scene, camera);
-        }
-
         for (let i = 0; i < models.length; i++) {
             const model = models[i];
 
             // filter model by view visibility
             if (model.enabled) {
-                if (LODModelsCachedUtils.isLODModelCulled(model)) {
+                if (scene?.isCulledByLod(camera, model)) {
```
Comment (minggo) [inline cocos/rendering/custom/web-scene.ts]: Please don't use `?`.

File: native/cocos/scene/RenderScene.cpp [hunk 5]
```diff
@@ -57,16 +126,20 @@ void RenderScene::activate() {
 
 bool RenderScene::initialize(const IRenderSceneInfo &info) {
     _name = info.name;
+    _lodStateCache = new LodStateCache(this);
```
Comment (minggo) [inline native/cocos/scene/RenderScene.cpp]: Please use `ccnew`.

File: cocos/render-scene/core/render-scene.ts [hunk 6]
```diff
@@ -505,3 +523,224 @@ export class RenderScene {
...
+        return (camera.visibility & layer) === layer;
+    }
+
+    private _renderScene: RenderScene = null!;
+
+    /**
+     * @zh LOD使用的model集合；包含每个LODGroup的每一级LOD
+     * @en The collection of models used by LOD; Each LOD of each LODGroup.
+     */
+    private _modelsIncludeByLODGroup: Map<Model, ModelInfo> = new Map<Model, ModelInfo>();
```
Comment (minggo) [inline cocos/render-scene/core/render-scene.ts]: _modelsIncludeByLODGroup -> _modelsIncludedInLODGroup

Comment (minggo) [reply cocos/render-scene/core/render-scene.ts]: Or just `_modelsInLODGroup` is ok.

**Review by minggo — APPROVED** [review]

**Pull request merged**